tasks/sep/027-holocene-system-config-upgrade-and-init-multi-chain/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {console2 as console} from "forge-std/console2.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {Simulation} from "@base-contracts/script/universal/Simulation.sol";
+import {NestedSignFromJson as OriginalNestedSignFromJson} from "script/NestedSignFromJson.s.sol";
+import {DisputeGameUpgrade} from "script/verification/DisputeGameUpgrade.s.sol";
+import {CouncilFoundationNestedSign} from "script/verification/CouncilFoundationNestedSign.s.sol";
+import {VerificationBase, SuperchainRegistry} from "script/verification/Verification.s.sol";
+import {SystemConfigUpgradeEcotoneScalars} from "script/verification/SystemConfigUpgradeEcotoneScalars.s.sol";
+
+contract NestedSignFromJson is OriginalNestedSignFromJson, CouncilFoundationNestedSign {
+    string constant l1ChainName = "sepolia";
+    string constant release = "v1.8.0-rc.4";
+    string[4] l2ChainNames = ["op", "metal", "mode", "zora"];
+
+    SystemConfigUpgradeEcotoneScalars[] sysCfgUpgrades;
+
+    constructor() {
+        for (uint256 i = 0; i < l2ChainNames.length; i++) {
+            sysCfgUpgrades.push(new SystemConfigUpgradeEcotoneScalars(l1ChainName, l2ChainNames[i], release));
```

Comment:
Here we deploy a new `SystemConfigUpgradeEcotoneScalars` per chain, which each contains its own bindings to the appropriate SuperchainRegistry data.

Reply:
Each instance caches some information when it is created, and this cached data is used to verify that (aside from the Ecotone scalars) no data changed in the `SystemConfig` after the task is executed.

Reply:
Added these explanations into the source code itself, as comments.